File: app/src/main/java/com/example/sarahn/waterreminderapp/customsViews/ClsCustomTextView.java
package com.example.sarahn.waterreminderapp.customsViews;

import android.content.Context;
import android.graphics.Typeface;
import android.util.AttributeSet;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.TextView;

import com.example.sarahn.waterreminderapp.R;

/**
 * Created by SarahN on 6/4/2017.
 */
public class ClsCustomTextView extends TextView implements Animation.AnimationListener{
    public ClsCustomTextView(Context context) {
        super(context);
        Typeface face= Typeface.createFromAsset(context.getAssets(), "fonts/Fonty.ttf");
        this.setTypeface(face);
        Animation animation = AnimationUtils.loadAnimation(context, R.anim.text_slide_out);
        animation.setStartOffset(2500);
        startAnimation(animation);
        animation.setAnimationListener(this);
    }

    public ClsCustomTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
        Typeface face= Typeface.createFromAsset(context.getAssets(), "fonts/Fonty.ttf");
        this.setTypeface(face);

        Animation animation = AnimationUtils.loadAnimation(context, R.anim.text_slide_out);
        animation.setStartOffset(2500);
        startAnimation(animation);
        animation.setAnimationListener(this);
    }

    public ClsCustomTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        Typeface face= Typeface.createFromAsset(context.getAssets(), "fonts/Fonty.ttf");
        this.setTypeface(face);
        Animation animation = AnimationUtils.loadAnimation(context, R.anim.text_slide_out);
        animation.setStartOffset(2500);
        startAnimation(animation);
        animation.setAnimationListener(this);
    }


    @Override
    public void onAnimationStart(Animation animation) {

    }

    @Override
    public void onAnimationEnd(Animation animation) {
            setVisibility(View.GONE);
    }

    @Override
    public void onAnimationRepeat(Animation animation) {

    }
}

File: app/src/main/java/com/example/sarahn/waterreminderapp/customsViews/ClsCustomTextView.kt
package com.example.sarahn.waterreminderapp.customsViews

import android.content.Context
import android.graphics.Typeface
import android.util.AttributeSet
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.TextView
import com.example.sarahn.waterreminderapp.R

/**
 * Created by SarahN on 6/4/2017.
 */
class ClsCustomTextView : TextView, Animation.AnimationListener {
    constructor(context: Context) : super(context) {
        val face = Typeface.createFromAsset(context.assets, "fonts/Fonty.ttf")
        this.typeface = face
        val animation = AnimationUtils.loadAnimation(context, R.anim.text_slide_out)
        animation.startOffset = 2500
        startAnimation(animation)
        animation.setAnimationListener(this)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        val face = Typeface.createFromAsset(context.assets, "fonts/Fonty.ttf")
        this.typeface = face
        val animation = AnimationUtils.loadAnimation(context, R.anim.text_slide_out)
        animation.startOffset = 2500
        startAnimation(animation)
        animation.setAnimationListener(this)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        val face = Typeface.createFromAsset(context.assets, "fonts/Fonty.ttf")
        this.typeface = face
        val animation = AnimationUtils.loadAnimation(context, R.anim.text_slide_out)
        animation.startOffset = 2500
        startAnimation(animation)
        animation.setAnimationListener(this)
    }

    override fun onAnimationStart(animation: Animation) {}
    override fun onAnimationEnd(animation: Animation) {
        visibility = GONE
    }

    override fun onAnimationRepeat(animation: Animation) {}
}